app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __setattr__(self, attr: str, value: int) -> None:
+        if attr == "health" and value <= 0 and self in Animal.alive:
+            Animal.alive.remove(self)
+        super().__setattr__(attr, value)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> bool:
+        self.hidden = not self.hidden
+        return
```

Comment:
what's this for?

Reply:
Herbivore has a method of hide, which changes the hidden property of the beast to the opposite value and helps to hide from carnivores.